include/fastdds/dds/core/condition/GuardCondition.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file GuardCondition.hpp
+ *
+ */
+
+#ifndef _FASTDDS_GUARD_CONDITION_HPP_
+#define _FASTDDS_GUARD_CONDITION_HPP_
+
+#include <fastdds/dds/core/condition/Condition.hpp>
+#include <fastrtps/fastrtps_dll.h>
+#include <fastrtps/types/TypesBase.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+/**
+ * @brief The GuardCondition class is a specific Condition whose trigger_value is completely under the control
+ * of the application.
+ *
+ */
+class GuardCondition : public Condition
+{
+public:
+
+    // GuardCondition not implemented.
+
+    /**
+     * @brief Set the trigger_value
+     * @param value new value for trigger
+     * @return RETURN_OK
+     */
+    RTPS_DllAPI eprosima::fastrtps::types::ReturnCode_t set_trigger_value(
```

Comment:
In general, in all the other files in the library that have the namespace `eprosima::fastdds::dds`, ReturnCode_t it is declared as `using`.

Reply:
:+1:

Reply:
This is what I meant in this comment:

https://github.com/eProsima/Fast-DDS/blob/09a7ddc94c79800c0f580cba3ee57fc5bb5eae4c/src/cpp/fastdds/domain/DomainParticipantImpl.hpp#L37